src/crawlers/cheerio_crawler.js

```diff
@@ -589,7 +589,7 @@ class CheerioCrawler {
 
             // It's not a JSON so it's probably some text. Get the first 100 chars of it.
             throw new Error(`${statusCode} - Internal Server Error: ${body.substr(0, 100)}`);
-        } else if (type === 'text/html' || type === 'application/xhtml+xml' || type === 'application/xml') {
+        } else if (type === 'text/html' || type === 'application/xhtml+xml' || type === 'text/xml' || type === 'application/xml') {
```

Comment:
This is getting a bit long. Let's make two constants instead of one. `HTML_AND_XML_MIME_TYPES` and `APPLICATION_JSON_MIME_TYPE`. Then use a concatenation of those two in setting the default mime types and use the first one here in this condition.